### PR TITLE
Authenticate users

### PR DESCRIPTION
**Pull request by masonforest**
We now authenticate users before authorizing them. This restricts controllers to signed in users only by default.

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -26,6 +26,11 @@ class ApplicationController < ActionController::Base
 
   protect_from_forgery
   skip_before_action :verify_authenticity_token, if: :json_request?, only: [:create, :acs]
+  before_action :authenticate_user
+
+  def authenticate_user
```
Comment (stackus) [inline app/controllers/application_controller.rb]: This is better named require_user. It does not actually authenticate the user.

Comment (masonforest) [inline app/controllers/application_controller.rb]: Good call -> ee8918ded4e19a9d73bd182513e90ccd9550d453

File: app/serializers/product_serializer.rb [hunk 2]
```diff
@@ -10,11 +10,13 @@
...
 #  deleted_at           :datetime
+#  product_type_id      :integer
 #  setup_price          :decimal(10, 4)   default(0.0)
 #  hourly_price         :decimal(10, 4)   default(0.0)
 #  monthly_price        :decimal(10, 4)   default(0.0)
-#  provisioning_answers :jsonb
-#  product_type         :string
+#  provisionable_type   :string
+#  provisionable_id     :integer
+#  provisioning_answers :json
```
Comment (chriskacerguis) [inline app/serializers/product_serializer.rb]: @masonforest What's the thought behind json vs jsonb?

Comment (masonforest) [inline app/serializers/product_serializer.rb]: Oops `rake db:migrate` made these changes. Good catch. Reverted.

**Comment by chriskacerguis** [comment]
:+1:

**Pull request merged**